### PR TITLE
Longw/telegraf upgrade

### PR DESCRIPTION
**Pull request by wanlonghenry**
telegraf upgrade, upgrade version from 1.24.2 to 1.25.1
pulling changes from upgrade for Scrape Pods Method

### CODE REVIEW

File: kubernetes/container-azm-ms-agentconfig.yaml [hunk 1]
```diff
@@ -73,13 +73,28 @@ data:
...
-        # When monitor_kubernetes_pods = true, replicaset will scrape Kubernetes pods for the following prometheus annotations:
-        # - prometheus.io/scrape: Enable scraping for this pod
-        # - prometheus.io/scheme: If the metrics endpoint is secured then you will need to
-        #     set this to `https` & most likely set the tls config.
-        # - prometheus.io/path: If the metrics path is not /metrics, define it with this annotation.
-        # - prometheus.io/port: If port is not 9102 use this annotation
-        monitor_kubernetes_pods = false
+        ## Enable replicaset scraping of k8s pods. Further settings as to which pods to scape are determined by the 'method' option below. When enabled, the default is to use annotations to determine whether to scrape or not.
+        # monitor_kubernetes_pods = false
+
```
Comment (ganga1980) [inline kubernetes/container-azm-ms-agentconfig.yaml:78]: I think, we can just upgrade telegraf  as part of this update .Lets not make config updates considering the metrics will be supported through metrics addon and we dont have customers are asking for these features.  For adding the config changes, will have to test all the config changes and scale etc.

File: build/linux/installer/conf/telegraf.conf [hunk 2]
```diff
@@ -608,7 +608,7 @@
 #    Region = "$TELEMETRY_AKS_REGION"
 # [[inputs.procstat]]
 #  #name_prefix="container.azm.ms/"
-#  exe = "fluent-bit"
+#  exe = "td-agent-bit"
```
Comment (pfrcks) [inline build/linux/installer/conf/telegraf.conf:611]: is this correct? AFAIK the exe was renamed to fluent-bit

File: kubernetes/container-azm-ms-agentconfig.yaml [hunk 3]
```diff
@@ -52,7 +52,6 @@ data:
           # if enabled will also stitch together container logs split by docker/cri due to size limits(16KB per log line)
           # enabled = "false"
 
-
```
Comment (pfrcks) [inline kubernetes/container-azm-ms-agentconfig.yaml:55]: extraneous change

File: kubernetes/container-azm-ms-agentconfig.yaml (hunk 1 above)
Comment (wanlonghenry) [reply kubernetes/container-azm-ms-agentconfig.yaml]: updated code

File: kubernetes/container-azm-ms-agentconfig.yaml (hunk 3 above)
Comment (wanlonghenry) [reply kubernetes/container-azm-ms-agentconfig.yaml]: changed

File: kubernetes/linux/setup.sh [hunk 4]
```diff
@@ -57,10 +57,10 @@ sudo apt-get install jq=1.5+dfsg-2 -y
 #used to setcaps for ruby process to read /proc/env
 sudo apt-get install libcap2-bin -y
 
-wget https://dl.influxdata.com/telegraf/releases/telegraf-1.24.2_linux_$ARCH.tar.gz
-tar -zxvf telegraf-1.24.2_linux_$ARCH.tar.gz
+wget https://dl.influxdata.com/telegraf/releases/telegraf-1.25.1_linux_$ARCH.tar.gz
```
Comment (jatakiajanvi12) [inline kubernetes/linux/setup.sh:60]: Can we move this to a variable? So, we dont have to change in multiple places

**Review by jatakiajanvi12 — DISMISSED** [review]

File: kubernetes/linux/setup.sh (hunk 4 above)
Comment (wanlonghenry) [reply kubernetes/linux/setup.sh]: changed

**Review by ganga1980 — DISMISSED** [review]

**Review by ganga1980 — APPROVED** [review]

**Pull request merged**